### PR TITLE
Jethro/com 215/load testing

### PR DESCRIPTION
**Pull request by Jethro-Swanson**
Locust Load testing code, has feature branch merged in so that should be merged first

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -12,4 +12,5 @@ pyinstrument==5.0.0
 Pyjwt==2.9.0
 pymongo==3.12.3
 python-dotenv==1.0.1
-requests==2.32.3
\ No newline at end of file
+requests==2.32.3
+locust==2.32.3
```
Comment (gapy04) [inline requirements.txt]: Order this alphabetically.

File: README.md [hunk 2]
```diff
@@ -95,3 +95,11 @@ pull to dockerhub using default tag (latest)
...
 docker pull swanso15/cupboard_backend
 ```
+
+## Locust Load Testing
+
+run `locust` in the command line in the base Backend directory (where this file is found)
+
+enter 100 as the number of concurrent users and 1 as the number of users per second
+
+for host put the aws instace url/ip with the port ie http://2.4.5.7:6060 (this is not the actual ip)
```
Comment (gapy04) [inline README.md]: Fix typos. "instance" and "i.e."

File: locustfile.py
Comment (gapy04) [inline locustfile.py:1]: In general,
1. Replace all double quotes " with single quotes ' to follow pep8 standards (unless block comment).
2. Remove commented out logging.info if we are not actually using it.
3. Add a block comment for all function (def) in the following format:
```
"""
Description

Args:
    If any besides self
    arg1: description of the arg

Returns:
    What it returns if it does return something
"""
```

File: locustfile.py [hunk 3]
```diff
@@ -0,0 +1,239 @@
+from locust import HttpUser, task, between
+import os
+import logging
+from utils.env_helper import load_env_variables
```
Comment (gapy04) [inline locustfile.py:4]: Reorder imports.

It should always be python native libraries i.e. os and logging, followed by third party libraries such as locust, and then code from us (i.e. utils.env_helper).

File: locustfile.py
Comment (gapy04) [inline locustfile.py:1]: Run flake8 and fix any errors

**Review by gapy04 — CHANGES_REQUESTED** [review]

File: locustfile.py [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+class QuickstartUser(HttpUser):
+    global user_count
+    wait_time = between(1, 5)
+
+    """
+    all_api
+
+    Sends queries to all cupboard apis
+    """
```
Comment (gapy04) [inline locustfile.py]: Put the block comment inside the function (right under the `def all_api(self)`). Do this for the other block comment too.

File: locustfile.py [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+        self.user_id = user_count
+        self.access_token = ''
+
+        if (self.access_token == ''):
+            load_env_variables()  # Adds the .env values to environment to allow use
+
+            # Log in user
+            print_str = 'Logging User In as user: '
+            print_str += str(os.environ['LOAD_TEST_USERNAME'])
+            print_str += '_'+str(self.user_id)+'@gmail.com'
```
Comment (gapy04) [inline locustfile.py]: add a space between the operators here like we do in other places. Specifically, like `+ str(self.user_id)`

File: locustfile.py [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+            logging.info(print_str)
+            username = str(os.environ['LOAD_TEST_USERNAME'])
+            username += '_'+str(self.user_id)+'@gmail.com'
```
Comment (gapy04) [inline locustfile.py]: Same thing as above, add space between the operators.

**Review by gapy04 — CHANGES_REQUESTED** [review]

**Review by gregoryvm — APPROVED** [review]

**Review by gapy04 — APPROVED** [review]

**Pull request merged**